spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClientAutoConfiguration.java

```diff
@@ -96,17 +93,13 @@ public KubernetesDiscoveryProperties getKubernetesDiscoveryProperties() {
 	@Configuration
 	public static class KubernetesDiscoveryClientHealthIndicatorConfiguration {
 
-		private KubernetesDiscoveryClientHealthIndicatorInitializer indicatorInitializer;
-
-		@Autowired
-		public void setIndicatorInitializer(ApplicationEventPublisher applicationEventPublisher, PodUtils podUtils) {
-			this.indicatorInitializer = new KubernetesDiscoveryClientHealthIndicatorInitializer(podUtils,
-					applicationEventPublisher);
-		}
-
-		@PostConstruct
-		public void postConstruct() {
-			this.indicatorInitializer.initialize();
+		@Bean
+		public KubernetesDiscoveryClientHealthIndicatorInitializer indicatorInitializer(
+				ApplicationEventPublisher applicationEventPublisher, PodUtils podUtils) {
+			KubernetesDiscoveryClientHealthIndicatorInitializer indicatorInitializer = new KubernetesDiscoveryClientHealthIndicatorInitializer(
+					podUtils, applicationEventPublisher);
+			indicatorInitializer.initialize();
```

Comment:
If your bean implements `IntializingBean` you wouldn't need to call this explicitly.

Reply:
I didnt know about that, updated PR!